### PR TITLE
ET-3512 Creating buckets in AWS

### PR DESCRIPTION
**Pull request by jorge-cr-13**
## What 🕵️ 🔍

- Creating IONOS buckets in AWS

## References 📝 🔗

- [Jira task](https://xainag.atlassian.net/browse/)

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

**Review by Robert-Steiner — COMMENTED** [review]
what about moving the buckets into their own bucket like 

```
 organization/
 	account.hcl
 	/_common
 	/eu-central-1
 		region.hcl
 		/buckets
	 		/remote_config_repo
 			...
```
?

File: internal_tools/buckets/ionos_buckets/env.hcl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+locals {
+  aws_account_id = 690046978283
+  profile        = "AdministratorAccess-690046978283"
+  tags = {
+    Environment = "Organization"
```
[the remark this answers is not in the document]
Comment (jorge-cr-13) [reply internal_tools/buckets/ionos_buckets/env.hcl]: Yeah I think it sounds good! 👍🏻

**Comment by jorge-cr-13** [comment]
> > what about moving the buckets into a new organization folder
> > ```
> >  organization/
> >  	account.hcl
> >  	/_common
> >  	/eu-central-1
> >  		region.hcl
> >  		/buckets
> > 	 		/remote_config_repo
> >  			...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ?
> 
> @jorge-cr-13 can please take a look at this comment

Yeah sorry, I didn't see it. I'm going to make the changes on the structure and apply it.

File: organization/_common/s3.hcl [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  source = "git::https://github.com/terraform-aws-modules/terraform-aws-s3-bucket?ref=v3.6.0"
+}
+
+inputs = {
+  acl = "private"
+
+  versioning = {
+    enabled = true
+  }
+
```
Comment (Robert-Steiner) [inline organization/_common/s3.hcl:15]: to make them really private
```suggestion
  block_public_acls                = true
  block_public_policy              = true
  ignore_public_acls               = true
  restrict_public_buckets          = true
  attach_require_latest_tls_policy = true
```

File: organization/_common/s3.hcl [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+inputs = {
+  acl = "private"
+
+  versioning = {
+    enabled                          = true
+    block_public_acls                = true
+    block_public_policy              = true
+    ignore_public_acls               = true
+    restrict_public_buckets          = true
+    attach_require_latest_tls_policy = true
```
Comment (Robert-Steiner) [inline organization/_common/s3.hcl:18]: i think they are not part of the `versioning` map

```hcl
 versioning = {
    enabled                          = true
}
block_public_acls                = true
block_public_policy              = true
ignore_public_acls               = true
restrict_public_buckets          = true
attach_require_latest_tls_policy = true
```

**Review by Robert-Steiner — APPROVED** [review]
Nice!

**Pull request merged**